### DISCUSSION TITLE
caddy e duckdns in docker, improvvisamente non accedo più dall'esterno

### DISCUSSION THREAD

**Discussion by pillolola** [indomus/forum#2310] [Networking]
Buongiorno a tutti, è più di un anno che utilizzo HA core in docker su rpi 4 4gb ram, nello stesso rpi girano i container duckdns e caddy per poter accedere all'istanza dall'esterno; tutto ha funzionato bene finon ad un paio di giorni fa, ora riesco ad accedere solo dall'interno, e se provo dall'esterno ricevo come errore del browser (sulla porta 80) "ERR_CONNECTION_REFUSED".
ho provato a cancellare la cartella data relativa al voliume del container caddy e riseguire la guida di installazione ma senza successo;
dai log del container caddy ho: https://pastebin.com/n45rid3s
mentre il file Caddyfile è il seguente: https://pastebin.com/iA1SEMYi
Qualcuno sa indicarmi quale possa essere il problema e come poterlo risolvere?
grazie

**Comment by snakuzzo** [comment]
Spero tu sappia che stai usando ancora l'ambiente di staging di Let's Encrypt...sicuramente lo starai facendo per fare delle prove prima di richiedere il certificato definitivo.

Ad ogni modo...

1. L'errore che hai accedendo dall'esterno è chiaro ed è normale. Dall'esterno non devi accedere sulla porta 80 ma, per come hai impostato il tutto nel Caddyfile, devi accedere usando la 443.
2. Tornando ai log di Caddy, falliscono tutte le challenge...sia quella HTTP che quella HTTPS. Sicuro di aver configurato correttamente il port forwarding ?
- Per la challenge http-01 devi abilitare sul router il port forwarding dalla 80 esterna -> 80 interna del server su cui hai Caddy
- Per la challenge tls-alpn-01 devi abilitare sul router il port forwarding dalla 443 esterna -> 443 interna del server su cui hai Caddy
Ovviamente devi creare il container o in modalità `network host` oppure devi fare il bind delle porte in maniera corretta.

Detto ciò, se hai sempre usato la config che hai scritto, ti assicuro che non l'hai MAI fatto funzionare in maniera perfetta, visto che fino ad oggi hai usato l'ambiente di staging. Questo vorrebbe dire che non hai mai usato l'app companion ufficiale per accedere da fuori e da browser avrai sempre avuto il warning sul certificato non valido perché non rilasciato da una CA valida.

**Comment by pillolola** [comment]
Ciao Diego, grazie della risposta, se intendi usare l' ambiente di staging di Let's Encrypt, per il fatto che ho de-commentato "acme_ca https://acme-staging-v02.api.letsencrypt.org/directory" e commentato sia l'header che il comando di reverse all'istanza HA nel Caddyfile, allora si, ne sono consapevole (come anticipato sopra, ho provato a ripartire da zero con la guida) volevo evitare di bloccare con le varie prove la richiesta di nuovi certificati per il dominio.
Considera che il portforwarding sul router non è stato mai modificato da un anno a questa parte ed ha funzionato fino a qualche giorno fa. Per quanto riguarda l'errore di accesso, ho puntato alla porta 80 (che quando caddy funziona correttamente risponde con la frase  "Saluti dal Crew inDomus.it") solo per cercare di avere qualche info in più sull'errore. 
ovviamente prima del problema, la parte ora commentata ne Caddyfile era decommentata, al posto degli asterischi ovviamente ci sono il mio fqdn e la porta che uso per accedere al mio HA e chiaramente commentato/eliminato  la parte "acme_ca https://acme-staging-v02.api.letsencrypt.org/directory".
Ho ricontrollato il portforwarding sul router e confermo di avere aperto le porte 80-443 e quella che uso per l'accesso dall'esterno di HA
Secondo te, come mai mi falliscono le chllenge? cosa altro posso controllare?

**Reply by snakuzzo** [reply]
Per favore fai reply sotto l'ultimo messaggio. Non creare nuovi thread come hai fatto ora.

Tornando al problema...
non puoi puntare la porta 80 per farti rispondere "Saluti da indomus".
Nel Caddyfile hai specificato che deve rispondere sulla 443...non vedi ?

```
https://***********.duckdns.org {
  respond "Saluti dal Crew inDomus.it"
  file_server
}
```

**Reply by snakuzzo** [reply]
Per l'errore sulle challenge...
1. il dominio non è aggiornato con l'associazione IP corretta
2. problema di port forwarding
Altro non mi viene in mente

**Reply by pillolola** [reply]
Hai ragione, parlavo di porta 80, ma in realtà intendevo 443 (usavo infatti https e non http).
Per quanto riguarda l'associazione dominio-ip, ho ricontrollato ora ed è correttamente associato.
Il port forwarding è attivo.
Posso con qualche prova fatta in precedenza, non volendo, aver bloccato il mio dominio alla richiesta di nuovi certificati?

**Reply by snakuzzo** [reply]
Dai log non mi pare siano stati superati i limiti.
Le cose quelle sono...assiucurati sempre di aver creato correttamente il container di Caddy (vedi i messaggi precedenti)

**Reply by pillolola** [reply]
per quanto riguarda il container, utilizzo docker compose, quindi la creazione del container non è cambiata  rispetto a qualche giorno fa in cui funzionava tutto, per chiarezza e controllo, qui il mio file compose: https://pastebin.com/naFdLjci
grazie ancora

**Reply by pillolola** [reply]
Come al solito snakuzzo centra e affonda il problema!!! ;-)
Per aiutare anche altri che possono trovarsi nella mia situazione incomprensibile!!!!
Effettivamente il problema era nel router, non so come mai nell'abilitazione delle porte (che nessuno ha mai modificato) pur vedendo tutto correttamente impostato, non veniva fatto il portforwarding corretto (nelle impostazioni vedevo il portforwarding ad esempio della porta 80 sull'ip interno di caddy, ma raggiungendolo dall'esterno rispondeva un altro dispositivo). 
Ho quindi eliminato tutte le regole portforwarding del router, riavviato lo stesso e riconfigurato il portforwarding.... e per magia tutto ha ripreso a funzionare.
Grazie ancora Diego.

**Reply by snakuzzo** [reply]
Bene dai! :)
